src/sonar.ts

```diff
@@ -21,26 +21,78 @@ export type SONAR_RESULT = "issues" | "hotspots";
 
 const MAX_PAGE_SIZE = 500;
 
-export async function retrieveSonarCloudIssues(
-  sonarCloudInputs: SonarCloudInputs,
+export async function retrieveSonarIssues(
+  sonarInputs: SonarInputs,
 ): Promise<SonarSearchIssuesResult> {
-  const url = buildSonarCloudIssuesUrl(sonarCloudInputs);
-  return retrieveSonarCloudResults(sonarCloudInputs, url, "issues");
+  const url = buildSonarIssuesUrl(sonarInputs);
+  return retrieveSonarResults(sonarInputs, url, "issues");
 }
 
-export async function retrieveSonarCloudHotspots(
-  sonarCloudInputs: SonarCloudInputs,
+export function buildSonarIssuesUrl({
+  sonarHost,
+  componentKey,
+}: SonarInputs): string {
+  const { prNumber } = getGitHubContext();
+  const path = "/api/issues/search";
```

Comment:
The leading `/` in this constant adds unwanted inflexibility.

It's possible that a SonarQube instance is hosted at some "context" instead of at the root of the HTTP URL (e.g. `https://sonar.example.pixee.ai/context`). The leading `/` in this and other path constants prevents the action from handling this case. If we simply omit the leading `/`, then we can handle both the common case and the case where SonarQube is served from some context.

See `URL` behavior in Node REPL:

```
> new URL("foo/bar", "https://sonar.io/context/")
URL {
  href: 'https://sonar.io/context/foo/bar',
  origin: 'https://sonar.io',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'sonar.io',
  hostname: 'sonar.io',
  port: '',
  pathname: '/context/foo/bar',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
> new URL("foo/bar", "https://sonar.io/")
URL {
  href: 'https://sonar.io/foo/bar',
  origin: 'https://sonar.io',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'sonar.io',
  hostname: 'sonar.io',
  port: '',
  pathname: '/foo/bar',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```